From 75310caa501973a2564538daee9c6ea339fdff0b Mon Sep 17 00:00:00 2001
From: mpaluri <mpaluri@qti.qualcomm.com>
Date: Tue, 8 Sep 2026 21:09:44 -0700
Subject: [PATCH] Set mapc_negotiation_mode in hostapd.conf

Set mapc_negotiation_mode to manual for Fronthaul VAPS in hostapd.conf

Signed-off-by: Manaswini Paluri <mpaluri@qti.qualcomm.com>

diff --git a/common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp b/common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp
index 6ab93f32d..84e26aa4e 100644
--- a/common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp
+++ b/common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp
@@ -5594,6 +5594,24 @@ bool ap_wlan_hal_nl80211::update_vap_credentials(
                 // we always need to get the mgmt frames (assoc req) for capability reports:
                 if (bss_it.fronthaul) {
                     conf.set_create_vap_value(vap_id, "notify_mgmt_frames", "1");
+                    /* Enable MAPC negotiation on fronthaul VAPs so hostapd participates in
+                     * EasyMesh coordinated TDMA discovery and capability exchange. */
+                    if (conf.get_vap_value(vap_id, "mapc_negotiation_mode") != "1") {
+                        if (conf.set_create_vap_value(vap_id, "mapc_negotiation_mode", "1")) {
+                            if (!conf.store("")) {
+                                LOG(WARNING) << "update_vap_credentials: failed to save "
+                                             << "mapc_negotiation_mode=1 in hostapd config for "
+                                             << m_radio_info.iface_name;
+                            } else {
+                                LOG(INFO) << "update_vap_credentials: set mapc_negotiation_mode=1 "
+                                          << "in hostapd config for " << m_radio_info.iface_name;
+                            }
+                        } else {
+                            LOG(WARNING) << "update_vap_credentials: failed to set "
+                                         << "mapc_negotiation_mode in hostapd config for "
+                                         << m_radio_info.iface_name;
+                        }
+                    }
                     if (get_hal_conf().certification_mode) {
                         // MBO and Interworking must be enabled on every fronthaul VAP so that
                         // the MBO IE and Interworking IE appear in beacons/probe-responses for
-- 
2.34.1

